File: Linear Search/Java/LinearSearch.java

class LinearSearch {
    public static int search(int arr[], int data) {
        int N = arr.length;
        for (int i = 0; i < N; i++) {
            if (arr[i] == data)
                return i;
        }
        return -1;
    }

    public static void main(String args[]) {
        int arr[] = { 12, 21, 8, 16, 34, 51, 6 };
        int data = 34;

        int result = search(arr, data);
        if (result == -1)
            System.out.print("Element is not found");
        else
            System.out.print("Element is found at index " + result);
    }
}

File: Linear Search/Python/linear_search.py

def linearSearch(arr, data):
    for i in range(0, len(arr)):
        if (arr[i] == data):
            return i
    return -1


arr = [12, 21, 8, 16, 34, 51, 6]
data = 18

result = linearSearch(arr, data)

if(result == -1):
    print("Element not found")
else:
    print("Element found at index", result)
